packages/dara-components/js/smart/chat/chat.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+
+import { StyledComponentProps, Variable, injectCss, useComponentStyles, useVariable } from '@darajs/core';
+import styled from '@darajs/styled-components';
+import { Message, Chat as UiChat } from '@darajs/ui-components';
+
+interface ChatProps extends StyledComponentProps {
+    /** Passthrough the className property */
+    className: string;
+    /** The value Variable to display and update */
+    value?: Variable<Message[]>;
+}
+
+const ChatWrapper = styled.div`
+    pointer-events: auto;
+
+    position: fixed;
+    z-index: 999;
+    right: 1rem;
+    bottom: -0.1rem;
+`;
+
+const ChatButton = styled.button`
+    width: 3.5rem;
+    height: 3.5rem;
+    margin: 0 2rem 2rem 0;
+    padding-top: 0.5rem;
+
+    color: ${(props) => props.theme.colors.background};
+
+    background-color: ${(props) => props.theme.colors.primary};
+    border: none;
+    border-radius: 3rem;
+
+    :hover {
+        background-color: ${(props) => props.theme.colors.primaryHover};
+    }
+
+    :active {
+        background-color: ${(props) => props.theme.colors.primaryDown};
+    }
+`;
+
+const StyledChat = injectCss(UiChat);
+
+/**
+ * The Chat component switches between a chat button and a chat sidebar, allowing the user to interact with a chat.
+ *
+ * @param props the component props
+ */
+function Chat(props: ChatProps): JSX.Element {
+    const [style, css] = useComponentStyles(props);
+    const [value, setValue] = useVariable(props.value);
+    const [showChat, setShowChat] = React.useState(false);
+
+    return (
+        <ChatWrapper>
+            {showChat && (
+                <StyledChat
+                    $rawCss={css}
+                    className={props.className}
+                    onClose={() => setShowChat(false)}
+                    onUpdate={setValue}
+                    style={style}
+                    value={value}
+                />
+            )}
+            {!showChat && (
+                <ChatButton onClick={() => setShowChat(true)}>
+                    <svg fill="none" height="32" viewBox="0 0 32 32" width="32" xmlns="http://www.w3.org/2000/svg">
```

Comment:
This should probably use our theme colors so it works with dark mode / other themes